src/Appium.Net/Appium/Android/UiAutomator/AndroidUiSelector.cs

```diff
@@ -0,0 +1,477 @@
+            return this;
+        }
+
+        /// <summary>
+        /// Set the search criteria to match the widget by its instance number. The instance value must
+        /// be 0 or greater, where the first instance is 0. For example, to simulate a user click on the
+        /// third image that is enabled in a UI screen, you could specify a a search criteria where the
+        /// instance is 2, the className(String) matches the image widget class, and enabled(boolean) is
+        /// true.
+        /// Maps to the UiSelector.instance(int) method.
+        /// </summary>
+        /// <param name="instance">
+        /// The 0-indexed instance to match on
+        /// </param>
+        /// <remarks>https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#instance</remarks>
+        public AndroidUiSelector Instance(int instance)
+        {
+            _builder.AppendFormat(".instance({0})", instance.RequireIsPositive(nameof(instance)));
+            return this;
+        }
+
+        /// <summary>
+        /// Set the search criteria to match widgets that are long-clickable. Typically, using this search
+        /// criteria alone is not useful. You should also include additional criteria, such as text,
+        /// content-description, or the class name for a widget. If no other search criteria is specified,
+        /// and there is more than one matching widget, the first widget in the tree is selected.
+        /// Maps to the UiSelector.longClickable(int) method.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <remarks>https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#longclickable</remarks>
+        public AndroidUiSelector IsLongClickable(bool value)
+        {
+            _builder.AppendFormat(".longClickable({0})", value.ToString().ToLowerInvariant());
+            return this;
+        }
+
+        /// <summary>
+        /// Set the search criteria to match the package name of the application that contains the widget.
+        /// Maps to the UiSelector.packageName(String) method.
+        /// </summary>
+        /// <param name="packageName"></param>
+        /// <remarks>https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#packagename</remarks>
+        public AndroidUiSelector PackageNameEquals(string packageName)
+        {
+            _builder.AppendFormat(".packageName(\"{0}\")", packageName.RequireNotNull(nameof(packageName)));
+            return this;
+        }
+
+        /// <summary>
+        /// Set the search criteria to match the package name of the application that contains the widget.
+        /// Maps to the UiSelector.packageNameMatches(String) method.
+        /// </summary>
+        /// <param name="regex"></param>
+        /// <remarks>https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#packagenamematches</remarks>
+        public AndroidUiSelector PackageNameMatches(string regex)
```

Comment:
I'm not sure how using the Regex class would add safety here. Do you have a specific scenario in mind?

Other pros/cons for using the Regex class:
- Pro: it would add a bit of explicitness to the method signature if someone doesn't read the argument name
- Pro: you would get regex syntax highlighting when using the Regex ctor
- Con: since we don't actually use the Regex class for the matching, some options on the Regex constructor would be misleading for an end user. Example: in the following call, the RegexOption and timeout arguments would be completely ignored `uiSelector.PackageNameMatches(new Regex(@"myapp$", RegexOptions.ECMAScript, matchTimeout:TimeSpan.FromSeconds(3)))`

I lean toward keeping it a string-typed argument - I think it causes less confusion.

Reply:
I'm fine with the current implementation.